File: CodingSamples/Java/Runtime/IteratorTest/SimpleStack.java
import java.util.Iterator;

class SimpleStack<E> implements Iterable<E> {

	class Node {
		E element;
		Node below;

		Node(E item) {
			element = item;
			below = top;
		}
	}

	private Node top;

	public void push(E item) {
		top = new Node(item);
	}

	public E pop() {
		E item = top.element;
		top = top.below;
		return item;
	}

	public boolean empty() {
		return top == null;
	}

	public Iterator<E> iterator() {
		//instantiating an inner local anonymous class which implements 
		//the Iterator iterface
		return new Iterator<E>(){
			
			Node current = top;

			public boolean hasNext() {
				return current != null;
			}

			public E next() {
				E item = current.element;
				current = current.below;
				return item;
			}
		};
	}
}

File: CodingSamples/DotNet/Runtime/GenTypeTest/DemoApp/SimpleStack.cs
class SimpleStack<V>
{
    class Node
    {
        internal V Element;
        internal Node Below;
    }

    private Node top;

    public void Push(V item)
    {
        top = new Node { Element = item, Below = top };
    }

    public V Pop()
    {
        V item = top.Element;
        top = top.Below;
        return item;
    }

    public bool Empty()
    {
        return top is null;
    }
}